### PR TITLE
feat: lemmas about List.findIdx?

### PR DESCRIPTION
**Pull request by semorrison**
Blocked by:
- [x] #292 
- [x] ~~#243~~

### CODE REVIEW

**Review by fgdorais — CHANGES_REQUESTED** [review]
Proof optimization.

File: Std/Data/List/Lemmas.lean [hunk 1]
```diff
@@ -1441,6 +1441,74 @@ theorem findIdx_get?_eq_get_of_exists {xs : List α} (h : ∃ x ∈ xs, p x) :
     xs.get? (xs.findIdx p) = some (xs.get ⟨xs.findIdx p, xs.findIdx_lt_length_of_exists h⟩) :=
   get?_eq_get (findIdx_lt_length_of_exists h)
 
+  /-! ### findIdx? -/
+
+@[simp] theorem findIdx?_nil : ([] : List α).findIdx? p i = none := rfl
+@[simp] theorem findIdx?_cons :
```
[the remark this answers is not in the document]
Comment (semorrison) [reply Std/Data/List/Lemmas.lean:1447]: Doesn't fit on one line unless you break after the colon.

[the remark this answers is not in the document]
Comment (fgdorais) [reply Std/Data/List/Lemmas.lean:1447]: I meant to separate the theorems. Sorry for the confusion.

**Review by fgdorais — APPROVED** [review]

File: Std/Data/List/Lemmas.lean (hunk 1 above)
[the remark this answers is not in the document]
Comment (semorrison) [reply Std/Data/List/Lemmas.lean:1448]: Done.

**Pull request merged**